lib/mapObjects/CGTownInstance.h

```diff
@@ -132,7 +132,17 @@ class DLL_LINKAGE CGTownBuilding : public IObjectInterface
 	BuildingSubID::EBuildingSubID bType;
 };
 
-class DLL_LINKAGE COPWBonus : public CGTownBuilding
+class DLL_LINKAGE CGBaseTownBonus : public CGTownBuilding
```

Comment:
Do not alter class inheritance of serializable objects, it breaks savegame backward compatibility and is very difficult to work around.